library/core/src/char/convert.rs

```diff
@@ -155,6 +156,20 @@ impl From<char> for u128 {
     }
 }
 
+/// Map `char` with code point in U+0000..=U+00FF to byte in 0x00..=0xFF with same value, failing
+/// if the code point is greater than U+00FF.
+///
+/// See [`impl From<u8> for char`](../std/primitive.char.html#impl-From<u8>) for details on the encoding.
+#[stable(feature = "u8_from_char", since = "1.53.0")]
+impl TryFrom<char> for u8 {
+    type Error = TryFromIntError;
```

Comment:
Should we consider introducing a new error type here? I'm worried that stabilizing it on `TryFromIntError` would prevent us from adding a more specific error message in the future, not to mention wondering if we should be adding a more specific error message right now.

Looking at `TryFromIntError` it seems like the error message we'd output right now will always be "out of range integral type conversion attempted". But I could see us wanting to say something like "unicode character cannot be converted into a byte: out of range integral type conversion attempted".

Reply:
Great catch. I agree with this and think your proposed error message is much better than what we would have otherwise.